File: Eclipse/w7cpframe/src/clsPfSupport.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;


    class clsPfSupport
    {


        public int key = 0;
        public int js = 0;
        public int rx = 0;          //.. joint X directional restraint ..
        public int ry = 0;          //.. joint Y directional restraint ..
        public int rm = 0;          //.. joint Z rotational restraint ..



        public void initialise()
        {
            key = 0;
            js = 0;
            rx = 0;
            ry = 0;
            rm = 0;
        }

        public void setValues(int supportKey, int SupportNode, int RestraintX, int RestraintY, int RestraintMoment)
        {
            key = supportKey;
            js = SupportNode;
            rx = RestraintX;
            ry = RestraintY;
            rm = RestraintMoment;
        }


        public String sprint()
        {
            String s;

            s = "";
            s = s + String.format("%8d",key);
            s = s + String.format("%6d",js);
            s = s + String.format("%6d",rx);
            s = s + String.format("%6d",ry);
            s = s + String.format("%6d",rm);

            return s;
        }

        public void cprint()
        {
            System.out.println(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }


        public void fgetData(BufferedReader fp)
        {
            String s="";

            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            System.out.println(s);
            sgetData(s);

            System.out.println("... fgetData");
        }


        public void sgetData(String s)
        {
            String [] dataflds = new String[10]; //(0 To 9);
            int i, n;

            System.out.println("sgetData ...");
            System.out.println(s);

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }


            n = dataflds.length;
            for (i = 0; i < n; i++)
            {
                System.out.println(i + "<" + dataflds[i] + ">");
            }

            key = Integer.parseInt(dataflds[0]);
            js = Integer.parseInt(dataflds[1]);
            rx = Integer.parseInt(dataflds[2]);
            ry = Integer.parseInt(dataflds[3]);
            rm = Integer.parseInt(dataflds[4]);



            System.out.println("... sgetData");
        }




    } //class

File: SharpDevelop/VBasic/w7cpframe/clsPfSupport.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfSupport

#Const isUsingExcel = 0

    Public key As Integer
    Public js As Byte
    Public rx As Byte            '.. joint X directional restraint ..
    Public ry As Byte            '.. joint Y directional restraint ..
    Public rm As Byte            '.. joint Z rotational restraint ..

    Sub initialise()
        js = 0
        rx = 0
        ry = 0
        rm = 0
    End Sub

    Sub setValues(ByVal supportKey As Integer, ByVal SupportNode As Integer, ByVal RestraintX As Byte, ByVal RestraintY As Byte, ByVal RestraintMoment As Byte)
        key = supportKey
        js = SupportNode
        rx = RestraintX
        ry = RestraintY
        rm = RestraintMoment
    End Sub


    Function sprint() As String
        sprint = StrLPad(Format(key, "##0"), 8) _
                 & StrLPad(Format(js, "##0"), 6) _
                 & StrLPad(Format(rx, "##0"), 6) _
                 & StrLPad(Format(ry, "##0"), 6) _
                 & StrLPad(Format(rm, "##0"), 6)
    End Function

    Sub cprint()
        Console.WriteLine(sprint)
    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub


#If isUsingExcel = 1 Then

  Sub wbkPrint(dataRec As Range, RecNum As Byte)
    With dataRec
      .Offset(RecNum, 0).Value = key
      .Offset(RecNum, 1).Value = js
      .Offset(RecNum, 2).Value = rx
      .Offset(RecNum, 3).Value = ry
      .Offset(RecNum, 4).Value = rm
    End With
  End Sub

  Sub wbkRead(dataRec As Range, RecNum As Byte)
    With dataRec
      key = .Offset(RecNum, 0).Value
      js = .Offset(RecNum, 1).Value
      rx = .Offset(RecNum, 2).Value
      ry = .Offset(RecNum, 3).Value
      rm = .Offset(RecNum, 4).Value
    End With
  End Sub

#End If

End Class
